common/setup/platform/posix/setup.cpp

```diff
@@ -0,0 +1,32 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <common/setup.hpp>
+
+#include <signal.h>
+
+namespace mender {
+namespace common {
+namespace setup {
+
+void GlobalSetup() {
+	struct sigaction action;
+	action.sa_handler = SIG_IGN;
+
+	sigaction(SIGPIPE, &action, nullptr);
```

Comment:
Q: When do we ever get a `SIGPIPE` ? :thinking: 

And what about term, abort, etc?

Reply:
When the reading end of a pipe closes its end, and the other end tries to write to it. This is used in the tests, and is expected to come up if an Update Module crashes in the Download state. In this scenario the client is writing to the process through a pipe, so it's important that the signal is ignored so that we can get an error code instead.

I assume that the other signals will be handled elsewhere, because they may need more targeted treatment. Or maybe some of it can be added here, not sure yet. But it's anyway something I won't handle now, because it's not as clear cut as `SIGPIPE`.

Reply:
I saw it for the tests. Didn't know we actually did it for the `Download` stat, but now that you mention it :)